### PR TITLE
Adding more tests to the sharing API.

### PR DESCRIPTION
**Pull request by SergioBertolinSG**
- [x] Enhancements to pre and post conditions.
- [x] General function to create shares in preconditions.
- [x] Delete shares.
- [x] Get all shares of a user.

_Pending tasks for future pull requests:_
- [ ] Get all shares.
- [ ] Get shares from specific file or folder.
- [ ] Get info about a known share.
- [ ] More scenarios about updating shares.

### CODE REVIEW

File: build/integration/features/bootstrap/FeatureContext.php [hunk 1]
```diff
@@ -354,23 +357,35 @@ public function appIsEnabled($app) {
...
+		$previous_user = $this->currentUser;
+		$this->currentUser = "admin";
 		$this->creatingTheUser($user);
 		$this->userExists($user);
+		$this->currentUser = $previous_user;
 	}
 
 	public function deleteUser($user) {
+		$previous_user = $this->currentUser;
+		$this->currentUser = "admin";
```
Comment (DeepDiver1975) [inline build/integration/features/bootstrap/FeatureContext.php]: admin ???

File: build/integration/features/bootstrap/FeatureContext.php [hunk 2]
```diff
@@ -388,6 +403,7 @@ public function creatingTheUser($user) {
 
 		$this->response = $client->send($client->createRequest("POST", $fullUrl, $options));
 		$this->createdUsers[$user] = $user;
+		echo "Creating a user inside creatingTheuser\n";
```
Comment (DeepDiver1975) [inline build/integration/features/bootstrap/FeatureContext.php]: echo ???

File: build/integration/features/bootstrap/FeatureContext.php [hunk 3]
```diff
@@ -603,6 +620,38 @@ public function addingExpirationDate() {
 		PHPUnit_Framework_Assert::assertEquals(200, $this->response->getStatusCode());
 	}
 
+	public static function createTextFile($path, $filename){
+		$myfile = fopen("$path" . "$filename", "w") or die("Unable to open file!");
+		$txt = "ownCloud test text file\n";
+		fwrite($myfile, $txt);
```
Comment (DeepDiver1975) [inline build/integration/features/bootstrap/FeatureContext.php]: file_put_contents

File: build/integration/features/bootstrap/FeatureContext.php (hunk 1 above)
Comment (SergioBertolinSG) [inline build/integration/features/bootstrap/FeatureContext.php]: Admin user is forced in those auxiliary functions which are not used from the scenario steps and require admin permissions.

File: build/integration/features/sharing-v1.feature [hunk 4]
```diff
@@ -58,3 +58,31 @@ Feature: sharing
     And the HTTP status code should be "200"
     And Public shared file "welcome.txt" with password "publicpw" can be downloaded
 
+  Scenario: getting all shares of a user
+    Given user "user0" exists
+    And user "user1" exists
+    And file "textfile0.txt" from user "user0" is shared with user "user1"
+    And As an "admin"
+    When sending "GET" to "/apps/files_sharing/api/v1/shares"
+    Then the OCS status code should be "100"
```
Comment (DeepDiver1975) [inline build/integration/features/sharing-v1.feature]: The content of the response has to be verified.

Especially: user 'admin' should not see the shared file. user1 should see the file

File: build/integration/features/bootstrap/FeatureContext.php [hunk 5]
```diff
@@ -603,6 +619,119 @@ public function addingExpirationDate() {
...
+	 */
+	public function deletingLastShare(){
+		$share_id = $this->lastShareData->data[0]->id;
+		$url = "/apps/files_sharing/api/v{$this->apiVersion}/shares/$share_id";
+		$this->sendingToWith("DELETE", $url, null);
+	}
+
+	public static function removeFile($path, $filename){
+		if (file_exists("$path" . "$filename")) {
+        	unlink("$path" . "$filename");
```
Comment (MorrisJobke) [inline build/integration/features/bootstrap/FeatureContext.php]: Indentation off

**Comment by DeepDiver1975** [comment]
@SergioBertolinSG I'd like to get this merged in the current state - please implement further tests in another branch - THX

**Comment by DeepDiver1975** [comment]
@MorrisJobke any objections to the code besides indentation?

@SergioBertolinSG please get PHPStorm - formating and stuff like that will be done automatically we don't need to comment on that on each pull request - THX

**Comment by SergioBertolinSG** [comment]
Please wait a little, I am currently making some changes.

**Comment by DeepDiver1975** [comment]
> Please wait a little, I am currently making some changes.

ok - no worries - but I need this to be done by this afternoon

File: build/integration/features/bootstrap/FeatureContext.php [hunk 6]
```diff
@@ -603,6 +619,143 @@ public function addingExpirationDate() {
...
 	}
 
+	public function createShare($user,
+								$path = null, 
+								$shareType = null, 
+								$shareWith = null, 
+								$publicUpload = null, 
+								$password = null, 
+								$permissions = null){
+		$fullUrl = $this->baseUrl . "v{$this->apiVersion}.php/apps/files_sharing/api/v{$this->apiVersion}/shares";
```
Comment (rullzer) [inline build/integration/features/bootstrap/FeatureContext.php]: This looks wrong. If you use v2 of the OCS API (so proper HTTP status codes) you should still use v1 of the SHARE API. We only have 1 version at the moment.

File: build/integration/features/bootstrap/FeatureContext.php [hunk 7]
```diff
@@ -603,6 +619,143 @@ public function addingExpirationDate() {
...
+			}
+		}
+		return False;
+	}
+
+	/**
+	 * @Given /^file "([^"]*)" from user "([^"]*)" is shared with user "([^"]*)"$/
+	 */
+	public function assureFileIsShared($filepath, $user1, $user2){
+		$fullUrl = $this->baseUrl . "v{$this->apiVersion}.php/apps/files_sharing/api/v{$this->apiVersion}/shares" . "?path=$filepath";
```
Comment (rullzer) [inline build/integration/features/bootstrap/FeatureContext.php]: same here

File: build/integration/features/bootstrap/FeatureContext.php (hunk 6 above)
Comment (rullzer) [inline build/integration/features/bootstrap/FeatureContext.php]: Ah because API version is set to 1 this works. If you set it to 2 it fails.

**Comment by rullzer** [comment]
Looks good in general (beside my comment). If that is fixed :+1:

**Comment by rullzer** [comment]
Maybe for future stuff the featurecontext should be split up. The file is getting pretty large already ;)

**Comment by DeepDiver1975** [comment]
:+1:

**Pull request merged**